src/game_objects/grid.cpp

```diff
@@ -75,6 +77,7 @@ void Grid::build(ConstructType type, int col, int row) {
     case CHEST: {
         Construct* to_add = new Crate(col, row);
         to_add->setup_model();
+        children.push_back(to_add);
```

Comment:
Either way was fine. The only reason why I'd slightly prefer having it a child of the grid is to remove the possible assumption that they are constrained to a single tile.